### PR TITLE
Lead window function

### PR DESCRIPTION
**Pull request by Chopikashvili**
### Description

Created the write-up for the LEAD() function, creating the `terms` branch, the `lead` branch and the `lead.md` file. Closes issue #1778.

### Type of Change

- Adding a new entry

### Checklist

- [*] All writings are my own.
- [*] My entry follows the Codecademy Docs style guide.
- [*] My changes generate no new warnings.
- [*] I have performed a self-review of my own writing and code.
- [*] I have checked my entry and corrected any misspellings.
- [*] I have made corresponding changes to the documentation if needed.
- [*] I have confirmed my changes are not being pushed from my forked `main` branch.
- [*] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [*] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (Chopikashvili) [reply content/sql/concepts/window-functions/terms/lead/lead.md]: Applied changes.

**Review by SSwiniarski — APPROVED** [review]
@Chopikashvili looks good for a second review.

**Review by caupolicandiaz — APPROVED** [review]
@Chopikashvili LGTM

**Comment by Chopikashvili** [comment]
I'm trying to get the tests to run. How could I solve this?

**Comment by yangc95** [comment]
> I'm trying to get the tests to run. How could I solve this?

This may be slightly out-of-scope as a contributor but the tests make for a good learning experience. See in the contribution guide under [I submitted a PR but tests are failing, how do I fix them?](https://github.com/Codecademy/docs/blob/main/.github/CONTRIBUTING.md#i-submitted-a-pr-but-tests-are-failing-how-do-i-fix-them) for some tips to passing these tests (I ran the script already and pushed up the change).

Also see [package.json](https://github.com/Codecademy/docs/blob/main/package.json) for other available script commands if you're curious.

**Comment by Chopikashvili** [comment]
Is two positive reviews enough for approving the changes? If so, provided the tests are fine, how long do I need to wait until the PR is merged?

**Comment by SSwiniarski** [comment]
@yangc95, were you wanting to review this, or can we just merge?

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (yangc95) [inline content/sql/concepts/window-functions/terms/lead/lead.md:27]: Can we link out to the order by entry: https://www.codecademy.com/resources/docs/sql/commands/order-by

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (yangc95) [inline content/sql/concepts/window-functions/terms/lead/lead.md:36]: Can we make this a bullet point along with what's written on line 27 (let's create a few points to explain the different parameters here)

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (yangc95) [inline content/sql/concepts/window-functions/terms/lead/lead.md:21]:
```suggestion
LEAD(column1_name, offset) OVER(
```

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (yangc95) [reply content/sql/concepts/window-functions/terms/lead/lead.md:24]: Let's isolate the LEAD() function from the rest of the query and focus on just that

**Review by yangc95 — COMMENTED** [review]
@Chopikashvili great example! At minimum, a new entry should receive two reviews.

I did take another look at this PR and I have some comments. Please let me know if you need any assistance or have any questions. 🙂

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (Chopikashvili) [reply content/sql/concepts/window-functions/terms/lead/lead.md]: Applied changes.

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (Chopikashvili) [reply content/sql/concepts/window-functions/terms/lead/lead.md]: Applied changes.

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (Chopikashvili) [reply content/sql/concepts/window-functions/terms/lead/lead.md]: Applied those and above changes.

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (yangc95) [reply content/sql/concepts/window-functions/terms/lead/lead.md:16]: That's fair, I am curious as to how you're testing out these queries.

I'm also wondering if, when looking into this function, that the syntax pertains to SQL, T-SQL, or both.

Comment (Chopikashvili) [reply content/sql/concepts/window-functions/terms/lead/lead.md]: Update: I was wrong. Applied changes.

**Comment by Chopikashvili** [comment]
I think everything should be good now.

**Comment by Chopikashvili** [comment]
Can someone make the tests run?

**Comment by Chopikashvili** [comment]
I'm very sorry. Can someone make the tests run and merge this?

**Comment by yangc95** [comment]
@Chopikashvili that’s alright, appreciate the momentum!

I can run the tests, but I’d like to ask that we add a note at the end of the syntax section to this entry that includes which SQL Server this syntax applies to.

**Comment by Chopikashvili** [comment]
@yangc95 All the popular servers have the same syntax for `LEAD()`.

**Comment by Chopikashvili** [comment]
Assuming servers are various DBMS's.

File: content/sql/concepts/window-functions/terms/lead/lead.md
Comment (yangc95) [inline content/sql/concepts/window-functions/terms/lead/lead.md:51]:
```suggestion
Running the following query shows the next month's sales for every month, with the last month having `NA as its value.
```

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
🥳 Thanks for your cooperation throughout this process @Chopikashvili! live link: https://www.codecademy.com/resources/docs/sql/window-functions/lead

Screenshot 2023-04-07 at 12 44 22 PM

> It looks as if you're anonymous. If you want to be credited follow the instructions below to add your GitHub username to your Codecademy profile. If not, there is nothing more you need to do.
> 
> 
> To add your GitHub username to your Codecademy profile:
> 
> 1.) Add your GitHub Username to your Codecademy Profile Details, you can get there on your desktop by pressing on your profile picture in the upper right-hand corner and selecting "Profile" in the drop-down box:
> 
> 
> 
> 2.) Then log out and log back into Codecademy using your GitHub login:
> 
> githublogin
> 
> If the steps above do not work, it may be due to an email or local configuration issue.
> 
> In addition, we may reach out in the future if any adjustments to these entries are required.
>

@caupolicandiaz @SSwiniarski

**Pull request merged**
